Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2647,6 +2647,18 @@ public struct Moments<Scalar: TensorFlowFloatingPoint>: Differentiable {
 }
 
 public extension Tensor where Scalar: TensorFlowFloatingPoint {
+    /// Helper function that assess if `axis` is in the range `[-rank, rank)`, where `rank` is the rank of
+    /// the provided tensors.
+    @usableFromInline
+    internal func preconditionAxis(_ axis: Int) {
```

Comment:
We have a `preconditionAxis` in `basic.swift`, is this redeclaration neccessary because of the scope?

Reply:
Yaa, it is working even without that declaration I was not aware of that function

Reply:
If it works without a declaration in this file, then this redeclaration should be removed.

Reply:
Ohh! sorry, it is not working without a declaration. I basic.swift the function is declared first and then it is used.

Reply:
Is duplicating `preconditionAxis` actually necessary? What happens if you remove the duplicate declarations?
`preconditionAxis` is internal (not private), so it should be accessible from anywhere in the `TensorFlow` module.